sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/KotlinSensor.kt

```diff
@@ -89,22 +89,22 @@ class KotlinSensor(
             fileSystem.predicates().hasType(InputFile.Type.MAIN)
         )
 
-        val filesToAnalyze = fileSystem.inputFiles(mainFilePredicate).let { mainFiles ->
+        val filePartition: Pair<Iterable<InputFile>, Iterable<InputFile>> = fileSystem.inputFiles(mainFilePredicate).let { mainFiles ->
             if (canSkipUnchangedFiles(sensorContext)) {
                 LOG.debug("The Kotlin analyzer is running in a context where it can skip unchanged files.")
                 var totalFiles = 0
-                mainFiles
-                    .filter {
-                        totalFiles++
```

Comment:
It's kind of nice to have both parts of the partition now, as we can use `.size` on both parts and add them up instead of the ugly side-effect logic of incrementing a counter.